bots/psuteam7bot/prophet.js

```diff
@@ -37,183 +55,171 @@ prophet.doAction = (self) => {
-            return self.attack(attacking.x - self.me.x, attacking.y - self.me.y);
-        }
-        
-        //TODO store 'reclaimed' resources if capacity full and no enemies [future sprint obj]
-        //Otherwise, idle.
-        return;
+    if(attackable.length > 0)
+    {
+        let attacking = attackable[0];
+        self.log("Attacking " + combat.UNITTYPE[attacking.unit] + " at " + attacking.x + ", " +  attacking.y);
+        return self.attack(attacking.x - self.me.x, attacking.y - self.me.y);
     }
-    else if(self.role === "ATTACKER")
+    
+    self.log("DEFENDER prophet guarding at x: " + self.me.x + ", y: " + self.me.y);
+    return;
+}
+
+
+//ATTACKER Behavior
+prophet.takeAttackerAction = (self) => {
+    self.log("ATTACKER prophet " + self.id + " taking turn");
+
+    //If no base
+    if(self.base == null)
     {
-        self.log("ATTACKER prophet " + self.id + " taking turn");
-        if(self.potentialEnemyCastleLocation === null)
-        {
-            //TODO Set 'Potential enemy castle coordinates' for prophets that spawned/ created at a Church
-            //Maybe need pioneer to 'hold' enemy castle locations and have Churches they built store the value and communicate it to ATTACKER?
-            //Requires communication and modifying other units
+        //Set opposite of current coord as target
+        self.potentialEnemyCastleLocation = movement.getAttackerPatrolRoute(self.me, self.map);
+    }
 
-            //Get potential enemy castle locations
-            self.potentialEnemyCastleLocation = movement.getAttackerPatrolRoute(self.base, self.map);
-            self.target = self.potentialEnemyCastleLocation[[Math.floor(Math.random()*2)]];
 
+    //If no target
+    if(self.potentialEnemyCastleLocation === null)
+    {
+        //Get potential enemy castle locations if Castle didn't send signal
+        self.potentialEnemyCastleLocation = movement.getAttackerPatrolRoute(self.base, self.map);
+    }
+
+    if(self.target === null)
+    {     
+        self.target = self.potentialEnemyCastleLocation[0];
+    }
 
-            //TODO Set 'Rally point?'? (For amassing friendly forces before attacking as a group  [future sprint obj])
-        }
-        //TODO Set 'Potential enemy castle coordinates if allied group threshold is reached' [future sprint obj]
+    const visibleRobots = self.getVisibleRobots();
+    const attackable = combat.filterByAttackable(self, visibleRobots);
 
-        //Get visible robots seems expensive in performance, seems to use a lot of chess timer, especially when there are lots of robots nearby
-        //Possibly main cause of script timeout as the game goes on
-        //get it once and use filtered value
-        const visibleRobots = self.getVisibleRobots();
-        const attackable = combat.filterByAttackable(self, visibleRobots);
+    if(prophet.fleeBehavior(self, visibleRobots))
```

Comment:
We probably want to move this below the attacking option - we don't want to flee as soon as an enemy appears, we want to attack it if we can

Reply:
See how it behaves, will move it above attack if necessary